### ISSUE TITLE
`load_file` should work with pathlib

### ISSUE THREAD

**Issue by billbrod** [pynapple-org/pynapple#299]
Hi pynapple devs. When calling `pynapple.load_file` with pathlib.Path (as opposed to a string), pynapple raises an `AttributeError` because it tries to call `path.endswith`:

```python
import pynapple as nap
import pathlib

p = pathlib.Path('test_file.npz')
nap.load_files(p)
```

(This came up for me because of some code I'm writing for [nemos](https://github.com/flatironinstitute/nemos) to download data in our examples, using [pooch](https://www.fatiando.org/pooch/latest/))

An easy fix would be to just call `path.as_posix()` to convert pathlib.Path objects to string, but using the Path objects directly under the hood will probably make many things easier.

**Comment by vigji** [comment]
I would volunteer to refactor this transition from `os` to `Path` objects in a PR to `dev` if something like this can be done, but I don't know if there is an underlying decision to stick to `os` from developers @gviejo

**Comment by gviejo** [comment]
Hi Luigi, happy to review a PR for this.

**Comment by gviejo** [comment]
Solved in #316
